### PR TITLE
docs(pulsar sink): autogen docs

### LINKED ISSUE #16026: Convert the `pulsar` sink into auto-generated Cue documentation

### PR DESCRIPTION
**Pull request by davidhuie-dd**
Closes #16026

### CODE REVIEW

**Review by spencergilbert — APPROVED** [review]

File: src/sinks/pulsar.rs [hunk 1]
```diff
@@ -47,9 +47,11 @@ enum BuildError {
 pub struct PulsarSinkConfig {
     /// The endpoint to which the Pulsar client should connect to.
     #[serde(alias = "address")]
+    #[configurable(metadata(docs::examples = "pulsar://127.0.0.1:6650"))]
     endpoint: String,
```
Comment (neuronull) [inline src/sinks/pulsar.rs:51]: Are there specific formats of the string which are acceptable, or required?
For example, is a port required? could a hostname work in addition to an IP address?
Is "pulsar://" scheme required in the string?
In other components, we have outlined the constraints a bit. 
Including different formats in additional examples can also help.

File: src/sinks/pulsar.rs [hunk 2]
```diff
@@ -78,6 +80,8 @@ pub struct PulsarSinkConfig {
     pub acknowledgements: AcknowledgementsConfig,
 
     /// Log field to use as Pulsar message key.
+    #[configurable(metadata(docs::examples = "message"))]
+    #[configurable(metadata(docs::examples = "my_field"))]
     partition_key_field: Option<String>,
```
Comment (neuronull) [inline src/sinks/pulsar.rs:85]: just noticing the generate config function sets this to "message".
But it looks like internally it truly is optional. Just thought it odd there is a default in the generate config.

File: src/sinks/pulsar.rs (hunk 1 above)
Comment (davidhuie-dd) [reply src/sinks/pulsar.rs:51]: Let me look into this.

Comment (davidhuie-dd) [reply src/sinks/pulsar.rs]: I added docs about this.

File: src/sinks/pulsar.rs (hunk 2 above)
Comment (davidhuie-dd) [reply src/sinks/pulsar.rs:86]: Yeah, but I think that's just used to test the toml serialization/deserialization?

Comment (neuronull) [reply src/sinks/pulsar.rs:86]: I think it's also used to generate default configs with the command "vector generate <...>"

File: src/sinks/pulsar.rs [hunk 3]
```diff
@@ -45,7 +45,8 @@ enum BuildError {
 #[configurable_component(sink("pulsar"))]
 #[derive(Clone, Debug)]
 pub struct PulsarSinkConfig {
-    /// The endpoint to which the Pulsar client should connect to.
+    /// The endpoint to which the Pulsar client should connect to. The endpoint should specify the
+    /// pulsar protocol and port.
```
Comment (neuronull) [inline src/sinks/pulsar.rs:49]: nit: we generally seem to be line breaking these. See

https://github.com/vectordotdev/vector/blob/2bfd63922e7d82f7616138b51251f27a3052c9b4/src/sources/splunk_hec/mod.rs#L62-L64

File: src/sinks/pulsar.rs (hunk 2 above)
Comment (davidhuie-dd) [reply src/sinks/pulsar.rs:87]: Updated this. Thanks.

**Review by neuronull — APPROVED** [review]

**Pull request merged**